### PR TITLE
Feat/use only id externally

### PR DESCRIPTION
**Pull request by mymphe**
This PR hides staking module indexes from the external interface. All external setters and getters use module ids now. Additionally, StakingRouter provides a getter for all module ids.

### CODE REVIEW

File: contracts/0.4.24/Lido.sol [hunk 1]
```diff
@@ -902,14 +905,12 @@ contract Lido is StETHPermit, AragonApp {
...
+            uint256 shares2mint =
+                _totalRewards.mul(totalFee).mul(_getTotalShares())
                 .div(
-                    _getTotalPooledEther()
-                        .mul(precisionPoints)
-                        .sub(_totalRewards.mul(totalFee))
-            );
+                    (_getTotalPooledEther().mul(precisionPoints))
+                    .sub(_totalRewards.mul(totalFee))
+                );
```
Comment (skozin) [inline contracts/0.4.24/Lido.sol:913]: Why this formatting change? The result is much less readable.

Comment (mymphe) [reply contracts/0.4.24/Lido.sol]: That change pertains to a former base branch, `fix/typos-base-precision` which has already been merged to #482. This PR includes change only to StakingRouter and respective tests.

**Comment by TheDZhon** [comment]
Could you please update the PR to resolve conflicts?

**Comment by mymphe** [comment]
> Could you please update the PR to resolve conflicts?

Conflicts resolved.

**Review by krogla — APPROVED** [review]
lgtm, but it seems that storage read optimization is possible

**Pull request merged**